pkg/scans/checksrunner.go

```diff
@@ -156,15 +144,32 @@ func (c *ChecksRunner) CreateScanChecks(id string) error {
 		return err
 	}
 
+	// The checksCreated var should be never nil for the scans that are using
+	// async creation, in any case we check this in order to avoid panics.
+	checksCreated := 0
+	if scan.ChecksCreated != nil {
+		checksCreated = *scan.ChecksCreated
+	}
+
+	// Recheck if the scan was already finished by other worker since the last query.
+	if *scan.CheckCount == checksCreated {
+		// Scans was already created.
+		level.Warn(c.l).Log("ScanAlreadyCreated", id)
+		return err
+	}
+
+	if scan.TargetGroups == nil || len(*scan.TargetGroups) == 0 {
+		// Scans with no target groups should not be RUNNING.
+		level.Warn(c.l).Log("ScanWithNoTargetGroups", id)
+		return err
+	}
+
 	// The checktypes info used to create the checks of a scan should be fixed
 	// thus stored in the scan, so even if that info changes meanwhile the
 	// checks of the scan are being created the checks remain the same.
 	if scan.ChecktypesInfo == nil {
 		// Scans with no target groups should not be RUNNING.
-		status := service.ScanStatusFinished
-		scan.Status = &status
 		level.Warn(c.l).Log("ScanWithNoChecktypesInfo", id)
-		_, err = c.store.UpdateScan(sid, scan, []string{service.ScanStatusRunning})
 		return err
```

Comment:
Same here, I would rewrite this to return `nil`

Reply:
makes sense